node_modules_koding/bongo/lib/bongo.coffee

```diff
@@ -105,71 +120,123 @@ module.exports = class Bongo extends EventEmitter
 
   bound: require 'koding-bound'
 
-  defineModel:(name, model)->
+
+  defineModel: (name, model) ->
+
+    unless model.addBongo? this
+      return
+
+    if @models[name]
+      console.warn "Model #{name} defined multiple times!"
+      return name
+
     @models[name] = model
-    return  unless model.addBongo? this
+
     @emit 'newModel', name, model
-    model.on 'needsPopulated', (def, path)=>
-      path = if Array.isArray path then path else path.split '.'
-      names = JsPath.getAt def, path
+
+    model.on 'needsPopulated', (def, _path) =>
+
+      _path = if Array.isArray _path then _path else _path.split '.'
+      names = JsPath.getAt def, _path
+
       unless Array.isArray names
         names = [names]
-        JsPath.setAt def, path, names
-      names.forEach (name, i)=>
+        JsPath.setAt def, _path, names
+
+      names.forEach (name, i) =>
         if target = @models[name]
-          JsPath.setAt def, path.concat(i), target
-      @on 'newModel', (newName, newModel)->
+          JsPath.setAt def, _path.concat(i), target
+
+      @on 'newModel', (newName, newModel) ->
         if newName in names
-          JsPath.setAt def, path.concat(names.indexOf newName), newModel
+          JsPath.setAt def, _path.concat(names.indexOf newName), newModel
+
+    return name
+
+
+  registerModel: (path) ->
```

Comment:
it's individual model register helper, it's not registering models it only registers the model in given path. it's using in iterator below.